index.d.ts

```diff
@@ -0,0 +1,56 @@
+import { ActionResult, ActionType, ActionsType, VNode, View } from 'hyperapp'
+
+export as namespace hyperapp__fx
+
+// hyperapp
+
+export interface App {
+    <State, Actions>(
+        state: State,
+        actions: ActionsType<State, Actions>,
+        view: View<State, Actions>,
+        container: Element | null,
+    ): Actions
+}
+
+// @hyperapp/fx
+
+export type EffectTuple = [string, object]
+
+export type Effect = EffectTuple | Array<EffectTuple>
+
+export type Action = (name: string, data?: any) => EffectTuple
+
+export type EffectsConfig = {
+    [effectName: string]: (props: object, getAction: (name: string) => Action) => undefined
+}
+
+export function withFx(fxOrApp: EffectsConfig | App): App
+
+export function action(name: string, data?: any): EffectTuple
+
+export function frame(action: string): EffectTuple
+
+export function delay(duration: number, action: string, data?: any): EffectTuple
+
+export function time(action: string): EffectTuple
+
+export function log(...args: Array<any>): EffectTuple
+
+export function http(url: string, action: string, options?: object): EffectTuple
```

Comment:
Should `options` be defined to mention the special [`response` and `error` props](https://github.com/hyperapp/fx#http) we support? Or any of the standard [`fetch` API props](https://developer.mozilla.org/en-US/docs/Web/API/Window/fetch#Parameters)?

Reply:
That may be. However, this is my initial suggestion that makes your library workable with TS.